e2e/cluster/e2e_test.go

```diff
@@ -0,0 +1,159 @@
+	env.GenerateControllerLogs(ctx, testStartTime)
+})
+
+var _ = Describe("Cluster E2E", Ordered, func() {
+	Context("Cluster Happy Path 🤖", Ordered, func() {
+		It("should onboard remote cluster", func() {
+			By("onboarding remote cluster")
+			shared.OnboardRemoteCluster(ctx, adminClient, env.RemoteKubeConfigBytes, remoteClusterName, env.TestNamespace)
+		})
+		It("should have a cluster resource created", func() {
+			By("verifying if the cluster resource is created")
+			Eventually(func(g Gomega) bool {
+				err := adminClient.Get(ctx, client.ObjectKey{Name: remoteClusterName, Namespace: env.TestNamespace}, &greenhousev1alpha1.Cluster{})
+				g.Expect(err).ToNot(HaveOccurred())
+				return true
+			}).Should(BeTrue(), "cluster resource should be created")
+
+			By("verifying the cluster status is ready")
+			shared.ClusterIsReady(ctx, adminClient, remoteClusterName, env.TestNamespace)
+		})
+
+		It("should verify remote cluster objects", func() {
+			By("verifying the remote cluster version")
+			cluster := &greenhousev1alpha1.Cluster{}
+			err := adminClient.Get(ctx, client.ObjectKey{Name: remoteClusterName, Namespace: env.TestNamespace}, cluster)
+			statusKubeVersion := cluster.Status.KubernetesVersion
+			dc, err := remoteRestClient.ToDiscoveryClient()
+			Expect(err).NotTo(HaveOccurred(), "there should be no error creating the discovery client")
+			expectedKubeVersion, err := dc.ServerVersion()
+			Expect(err).NotTo(HaveOccurred(), "there should be no error getting the server version")
+			Expect(statusKubeVersion).To(Equal(expectedKubeVersion.String()))
+
+			By("verifying if the managed namespace exists in the remote cluster")
+			ns := &corev1.Namespace{}
+			err = remoteClient.Get(ctx, client.ObjectKey{Name: env.TestNamespace}, ns)
+			Expect(err).NotTo(HaveOccurred(), "there should be no error getting the managed namespace")
+			Expect(ns.Status.Phase).To(Equal(corev1.NamespaceActive), "managed namespace must be active")
+
+			By("verifying if the cluster role binding exists in the remote cluster")
+			crb := &rbacv1.ClusterRoleBinding{}
+			err = remoteClient.Get(ctx, client.ObjectKey{Name: shared.ManagedResourceName}, crb)
+			Expect(err).ToNot(HaveOccurred(), "there should be no error getting the cluster role binding")
+
+			By("verifying if the greenhouse service account exists in the remote cluster")
+			sa := &corev1.ServiceAccount{}
+			err = remoteClient.Get(ctx, client.ObjectKey{Name: shared.ManagedResourceName, Namespace: env.TestNamespace}, sa)
+			Expect(err).ToNot(HaveOccurred(), "there should be no error getting the service account")
+
+			By("verifying if the greenhouse service account is bound to the cluster role binding")
+			found := false
+			for _, subject := range crb.Subjects {
+				if subject.Kind == rbacv1.ServiceAccountKind && subject.Name == sa.Name && subject.Namespace == env.TestNamespace {
+					found = true
+					break
+				}
+			}
```

Comment:
This could be simplified by using [slices.ContainsFunc(..)](https://pkg.go.dev/slices#ContainsFunc)

Reply:
done